src/core/memory.d

```diff
@@ -809,3 +809,41 @@ struct GC
         gc_runFinalizers( segment );
     }
 }
+
+/// Pure variant of `malloc` that doesn't affect global variable `errno`.
+void* pureMalloc(size_t size) pure
+{
+    const errno = _pureGetErrno();
```

Comment:
Does druntime use this convention of prefixing private symbols with one underscore?